src/main/webapp/app/exercises/programming/manage/programming-exercise-detail.component.ts

```diff
@@ -88,12 +88,16 @@ export class ProgrammingExerciseDetailComponent implements OnInit, OnDestroy {
                     if (latestTemplateResult) {
                         this.programmingExercise.templateParticipation.results = [latestTemplateResult];
                     }
+                    // This is needed to access the exercise in the result details
+                    this.programmingExercise.templateParticipation.programmingExercise = this.programmingExercise;
```

Comment:
I suggest using Angular's `@Input` instead. Manipulating the model is quite error-prone. => Then also revert the other file in that regard

Reply:
I don't see that adding another input is less error-prone than fixing the association here. Note that this is for some reason only a problem for template and solution, and I assume the added lines are missing. We manipulate the model in many places (and need to), including the lines above and below these two. With the input, we would effectively have two exercises in the context.

Reply:
So the reason I prefer the Input here is that we don't have a dependency chain that could have been manipulated at another place and therefore break anything because we directly supply the exercise.

Might this also be something worth discussing? 👀 😅

Reply:
Well, at least the components changed here are read-only, they don't do updates on the server. The issue would be the same for the submission of the result and the participation, both of which are also indirectly referenced in the result component. And we have many, many more in the client. We can discuss this further in the team, but I see more problems arising from that than it solves. At least for now, I couldn't name any actual problem we would solve. But it is quite common in Artemis to have attributes available sometimes, and sometimes not. Refactoring that is a lot of work, I am not sure how beneficial that would turn out eventually / how realistic that is.

I would regard it at least out of scope for this PR.

Some additional info on what I've done:
- I did a bit of web search, but could not find that any approach was considered dangerous / preferred and why, but my searches may have missed something.
- I tried checking the only two possible "update" actions, which are deleting a result or submission.
  Both currently cause different bugs in Artemis for template and solution. Result deletion simply is not executed and the modal "hags", submission deletion causes a class cast exception on the server. Looks like no-one every noticed that they were broken.

Reply:
I see. Thanks for taking a look into it.
I expected the Input to have a proper check but that is not the case as we can see from the issue with the if clause.
So yeah, probably currently not worth digging into too much.

Could you maybe create issues for the two bugs you've found so they are documented properly?

Reply:
Done, see #4008 and #4009.